modules/dlm/src/main/java/org/elasticsearch/dlm/DataLifecycleService.java

```diff
@@ -63,24 +75,59 @@ public class DataLifecycleService implements ClusterStateListener, Closeable, Sc
         Setting.Property.Dynamic,
         Setting.Property.NodeScope
     );
+    /*
+     * This setting controls the number of non-snapshot-related forcemerge errors that can happen for an index before we give up retrying
+     * forcemerge on that index.
+     */
+    public static final Setting<Integer> DLM_MAX_FORCEMERGE_ERRORS_SETTING = Setting.intSetting(
+        "indices.dlm.max_forcemerge_errors",
+        5,
+        1,
+        Setting.Property.Dynamic,
+        Setting.Property.NodeScope
+    );
```

Comment:
I'd suggest we don't add any error tracking here as we should design a potential back-off on its own as opposed to piggy backing it on this PR (which is already quite large)

I'd argue we might want to track in-flight requests as opposed to errors, but as I said IMO we should do that separately.

@dakrone would you agree?

Reply:
OK I've removed all of that and changed it so that it doesn't log an error if you get the same forcemerge exception repeatedly on an index. It will continue retrying forcemerge on an index indefinitely.